lecture_3/03_project/calculator/internal/engine/handler/handler.go

```diff
@@ -20,110 +20,86 @@ func New(betRepository BetRepository) *Handler {
 	}
 }
 
-// HandleBetsReceived handles bets received.
-func (h *Handler) HandleBetsReceived(
+// HandleBetsFromController handles bets from controller.
+func (h *Handler) HandleBetsFromController(
```

Comment:
Not sure where exactly to put the comment, but what I'd prefer changing in this function is determining whether the bet already exists in the repository - if not, insert it; if yes, update it.
Why? Because there can be more services in the system which are doing some operations on bets and then Controller would produce to `bets` queue even changes which were not made by Calculator. For example, let's say payment changes (for whichever reason) by some Payment service - that change is relevant for Calculator and it should handle the message accordingly.
Note that this will require you to add more methods to `BetRepository`.